addons/auth_signup/controllers/main.py

```diff
@@ -54,6 +55,11 @@ def web_auth_signup(self, *args, **kw):
                     _logger.error("%s", e)
                     qcontext['error'] = _("Could not create a new account.")
 
+        elif 'p_id' in qcontext:
+            partner = request.env['res.partner'].sudo().browse(int(qcontext.get('p_id')))
+            if all(u.state != 'new' for u in partner.user_ids):
+                return request.redirect('/web/login?login=%s' % url_quote(partner.email))
+
```

Comment:
🔥 with this you introduce a security issue

I can write a script without having any user to iterate on `/web/signup?p_id=XXXX` and retrieve all the partner emails in your database... (by just looking at the redirected URL)

So basically with this, all partner emails become public :/

When you display something on a route (especially for a portal / non-logged user), you should always think "if I'm a bad person, what can I do with this ?"

(same for public methods on model)


we already have a `signup_token` on the `res.partner` model, you  should probably check that it's valid